client/src/components/DisplayTable.js

```diff
@@ -64,12 +69,24 @@ export class DisplayTable extends React.Component {
             <tr className="table-header">
               <th 
                 className="center-text"
-                onClick={ () => this.header_click("label") }
               >
                 {label_col_header || ""}
-                <SortDirections 
-                  asc={!descending && sort_by === "label"} 
-                  desc={descending && sort_by === "label"}
+                <div
+                  onClick={ () => this.header_click("label") }
+                >
+                  <SortDirections 
+                    asc={!descending && sort_by === "label"} 
+                    desc={descending && sort_by === "label"}
+                  />
+                </div>
+                <input
+                  type="text"
+                  style={{width: "100%"}}
+                  value={activity_query}
+                  onChange={evt => 
+                    this.setState({ activity_query: evt.target.value })
+                  }
```

Comment:
I think it makes sense to assume the first column of a table is always filterable, but it might not always be activities (could be a table of departments). Can you rename this query to something like 'id_query' or 'first_column_query'?